### PR TITLE
`components`: API promotions

### PR DESCRIPTION
**Pull request by grigasp**
Part of https://github.com/iTwin/itwinjs-backlog/issues/1103.

@saskliutas, can you please have a look at  `packages\components\src\presentation-components\properties\editors\NavigationPropertyEditorContext.ts`? `navigationPropertyEditorContext` is exposed as public, although `createContext` results are generally internal elsewhere - I wasn't sure we want to make it `public` or `internal`.

### CODE REVIEW

**Review by saskliutas — APPROVED** [review]
I think `navigationPropertyEditorContext` should be internal. There is no reason it should be used outside NavigationPropertEditor.

**Comment by grigasp** [comment]
> I think `navigationPropertyEditorContext` should be internal. There is no reason it should be used outside NavigationPropertEditor.

Made the change, please review.

File: packages/components/src/presentation-components/properties/editors/NavigationPropertyEditorContext.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+export function NavigationPropertyEditorContextProvider({ children, ...props }: PropsWithChildren<NavigationPropertyEditorContextProviderProps>) {
+  return <navigationPropertyEditorContext.Provider value={props}>{children}</navigationPropertyEditorContext.Provider>;
+}
+
+/**
+ * Returns context provided by [[NavigationPropertyEditorContextProvider]]. The context is required for rendering [[NavigationPropertyEditor]] with editable
+ * navigation property values.
+ * @public
+ */
+export function useNavigationPropertyEditorContext() {
```
Comment (saskliutas) [inline packages/components/src/presentation-components/properties/editors/NavigationPropertyEditorContext.tsx:53]: I don't think this needs to be exported. It should not be used outside NavigationPropertyEditor.

**Review by saskliutas — CHANGES_REQUESTED** [review]

File: packages/components/src/presentation-components/properties/editors/NavigationPropertyEditorContext.tsx (hunk 1 above)
Comment (grigasp) [reply packages/components/src/presentation-components/properties/editors/NavigationPropertyEditorContext.tsx:54]: made `NavigationPropertyEditorContext` and `useNavigationPropertyEditorContext` internal

**Review by saskliutas — APPROVED** [review]

**Pull request merged**